ruby/vampires.rb

```diff
@@ -1,11 +1,55 @@
-puts "what is your name?"
-name = gets.chomp
+puts "how many employees will be processed?"
+number = gets.chomp.to_i
 
-puts "how old are you? what you were you born?"
-age_year = gets.chomp
+survey = 0
+while survey < number
+
+  puts "what is your name?"
+  name = gets.chomp 
+    
+  puts "how old are you? what year were you born?"
+  age_year = gets.chomp.to_i
+  puts "should we order some garlic bread for you?"
+    garlic_bread = gets.chomp
+    
+  puts "would you like to enroll in the company's health insurance?"
+  health_insurance = gets.chomp
+  
+  puts "list out any allergies. type done when finished."
+  allergies = gets.chomp
+  
+    until allergies == "done"
+     puts "List out any allergies type done when finished."
+     allergies = gets.chomp
+    end
+  
+  if age_year < 100 && garlic_bread == "yes" || health_insurance == "yes"
+    	puts "Probably not a vampire"
+  elsif age_year > 100 && garlic_bread == "no" 
+      puts "Probably a vampire"
+  elsif age_year > 100 && health_insurance == "no"
+      puts "Probably a vampire"
+  elsif age_year > 100 && garlic_bread == "no" && health_insurance == "no"
+      puts "Almost certainly a vampire"
+  else 
+    	age_year > 100 && garlic_bread == "yes" && health_insurance == "yes"
+  end 
+
+    
+  if name == "Drake Cula" || name == "Tu Fang"
```

Comment:
Let consider order of operations here.  I understand that the instructions may be guiding you to just put in conditions as they are mentioned.  However, we could clean this up and make the file much more efficient by analyzing our conditions for optimal calls and runtime.

If the employee is named "Drake Cula" or "Tu Fang" then we know that they are definitely a vampire. If that's the case, is there any need to even run the rest of the code?

Consider nesting your conditions and creating a strong order of operations for optimized logic and runtime.

```
if name == "Drake Cula" || name == "Tu Fang"
   "definitely a vampire"
else
  if age_check
    do something
  elsif garlic_and_insurance_check
    do something
  else
    do something
  end
end
```